test/GPv2SafeERC20.test.ts

```diff
@@ -159,7 +175,28 @@ describe("GPv2SafeERC20.sol", () => {
             recipient.address,
             amount,
           ),
-        ).to.be.revertedWith("GPv2SafeERC20: failed transfer");
+        ).to.be.revertedWith("failed transferFrom");
```

Comment:
This is why I dislike successful partial matches in revert messages.

Reply:
I like them.... :stuck_out_tongue:

Reply:
We can change this to `transfer failed` and `transferFrom failed` so the test would have caught the typo.

Reply:
Sorry, I didn't want to imply I wanted some change, I'm ok with how it is right now. Not something that can be done by us, but maybe a good compromise would be something like `.to.be.revertedMatching(/failed transferFrom/);`

Reply:
> .revertedMatching(/failed transferFrom/);

Did you mean `.revertedMatching(/failed transferFrom$/);`? Otherwise I think they are the same.

Reply:
Fur us, yes! But in general it'd look very similar to the current `revertedWith` but would be much more flexible without the extra "surprise" that partial matches are also accepted.
Anyway, I'm marking this as resolved since no changes are really needed.

Reply:
Ah I see. I thought `revertedMatching` already existed.